crates/types/src/traits/consensus_api.rs

```diff
@@ -13,7 +13,8 @@ use async_trait::async_trait;
 
 use std::{num::NonZeroUsize, time::Duration};
 
-/// The API that [`HotStuff`] needs to talk to the system
+/// The API that tasks use to talk to the system
+/// TODO do we plan to drop this?
```

Comment:
I think we discussed this at a team meeting recently. My summary is - we'd like to remove it, but it runs into `task-impls` and `hotshot` becoming dependent on each other. If I'm not missing something, we agreed that we can just demote `task-impls` to a submodule of `hotshot` and nuke this trait, but I didn't actually open an issue then. I went ahead and opened one: #2294

Reply:
Awesome, thanks for the summary! I went ahead and linked the issue here instead